### PR TITLE
75 make auth tests more generic

### LINKED ISSUE #75: make auth tests more generic
Something we've run into is that the auth tests in testing.py need to be made more generic, so that as other authenticators are used, we don't lose the value in that set of tests.
- they are integration tests, so as long as the selected authenticator works as expected (which should be verified in unit tests), we should be able to mock the authenticator's `verify_user`
- some tests won't be needed, depending on the backend
  - for instance, an LDAP-based system would likely not want forgot password, reset password, verify account, etc.

### PR DESCRIPTION
**Pull request by calebsyring**
fixes #75

### CODE REVIEW

**Comment by calebsyring** [comment]
Tests fail because `self.protected_url` points to a view function:
https://github.com/level12/keg-auth/blob/55ac405ca643724d81195036a7591444bc6f23d5/keg_auth_ta/views.py#L48
and not a class, so the `__keg_auth_requires_permissions__` obviously won't work.

So, a few different options jump out at me:

- remove the `secret1` function and just use the `Secret1Class`
- not use `__keg_auth_requires_permissions__` at all and leave it the way it was **OR** just make the permission testing users superusers

**Comment by guruofgentoo** [comment]
Couple of things here:
- decorated view functions will have the `__keg_auth_requires_permissions__` attribute directly, rather than on a `view_class` attribute, so it is still accessible in this case
- `__keg_auth_requires_permissions__`, though, is not necessarily a list that can be passed as permissions. It is more properly a list of conditions, the simplest of which is a permission token. It can also be a nested structure of https://github.com/level12/keg-auth/blob/master/keg_auth/model/utils.py#L4, or a callable, or some combination thereof.
  - If you want to specify for simplicity that the protected URL for testing must be protected only with permission tokens, that's one way to go
  - Or, traverse the structure for any tokens, knowing that you won't be able to support the callable cases easily
  - Or (this seems to be a simpler solution) was there an objection to having a testing-only route on the auth blueprint specifically for this sort of test? I appreciate the desire to make it more flexible, but anticipating all of the possible app-specific scenarios will be impossible, and will lead to further complexity down the road.

File: keg_auth/testing.py [hunk 1]
```diff
@@ -19,65 +20,72 @@ class AuthTests(object):
...
 
     def test_login_field_success_next_parameter(self):
         self.user_ent.testing_create(email='foo@bar.com', password='pass')
 
         next = '/foo'
         client = flask_webtest.TestApp(flask.current_app)
-        resp = client.get('{}?next={}'.format(self.login_url, next))
+        resp = client.get('{}?next={}'.format(
+            flask.url_for(flask.current_app.auth_manager.endpoint('login')), next
+            )
```
Comment (nZac) [inline keg_auth/testing.py]: This paren is kinda in a weird spot.

File: keg_auth/testing.py [hunk 2]
```diff
@@ -87,15 +95,19 @@ def test_login_field_success_next_session(self):
...
 
             resp.form['login_id'] = 'foo@bar.com'
             resp.form['password'] = 'pass'
             resp = resp.form.submit()
 
         assert resp.status_code == 302, resp.html
         assert resp.headers['Location'] == 'http://keg.example.com{}'.format(next)
-        assert resp.flashes == [('success', 'Login successful.')]
+        flash_success = flask.current_app.auth_manager.login_authenticator_cls.\
+            responder_cls['login'].flash_success
```
Comment (nZac) [inline keg_auth/testing.py]: In the tests its find to use `# noqa` to get around line length limits in Flake 8 This is a good example of where it is probably cleaner to do that.

File: keg_auth/testing.py [hunk 3]
```diff
@@ -19,65 +20,72 @@ class AuthTests(object):
     """
         These tests are designed so they can can be imported into an application's tests
         and ran to ensure customization of KegAuth hasn't broken basic functionality.
-
-        TODO: the messages we test for need to be configurable on the class in case the app
-        customizes then.  Ditto some of the redirect logic.
     """
-    login_url = '/login'
     protected_url = '/secret1'
```
Comment (nZac) [inline keg_auth/testing.py]: In order to make this generic add a class attribute which allows subclasses to add permissions to the user who is trying to hit the `proected_url` and add those to the user during `testing_create`.

**Pull request merged**